src/box/memtx_space.c

```diff
@@ -844,6 +858,13 @@ memtx_space_check_index_def(struct space *space, struct index_def *index_def)
 				 "BITSET can not be unique");
 			return -1;
 		}
+		if (index_def->opts.fast_offset) {
+			diag_set(ClientError, ER_MODIFY_INDEX,
+				 index_def->name, space_name(space),
+				 "BITSET index does not support "
+				 "logarithmic select with offset");
+			return -1;
+		}
```

Comment:
Instead of duplicating this check for each index type, let's just put it near the hint check below:

```C
	if (index_def->type != TREE && index_def->opts.fast_offset) {
		diag_set(ClientError, ER_MODIFY_INDEX, index_def->name,
			 space_name(space),
			 "fast_offset is only reasonable with memtx tree index");
		return -1;
	}
```

Reply:
Done.